dynamiq/nodes/agents/exceptions.py

```diff
@@ -59,3 +59,27 @@ def __init__(
         self, message: str = "Maximum number of loops reached without finding a final answer.", recoverable: bool = True
     ):
         super().__init__(message, recoverable=recoverable)
+
+
+class ParsingError(RecoverableAgentException):
+    """Base class for parsing errors."""
+
+    pass
+
+
+class XMLParsingError(ParsingError):
+    """Exception raised when XML structure is invalid or cannot be parsed."""
+
+    pass
+
+
+class MissingTagError(ParsingError):
```

Comment:
```suggestion
class TagNotFoundError(ParsingError):
```
to be consistent with other similar exceptions. Don't forget to fix docstrings too